Blog post for third-party cookie deprecation trial (#7828)
GoogleChrome/developer.chrome.com: site/en/blog/third-party-cookie-deprecation-trial/index.md

## Eligibility criteria and review process

This deprecation trial differs from previous trials with the introduction of a review and approval process for participation. This is to strike a balance between improving privacy for people on the web, while still enabling the services they depend on to request extra time to migrate if necessary.

The principles guiding this deprecation trial are: 

* **Preserving user-critical functionality:** This deprecation trial is intended for third-party providers that demonstrate functional breakage in user journeys.
* **Limiting user tracking:** The deprecation trial is not intended to support cross-site tracking for advertising purposes, and as such third-party embeds and services used for advertising are not eligible.

The ineligibility of advertising use cases will also help to ensure the deprecation trial does not interfere with the [industry testing planned for the start of 2024 as described by the Competition and Markets Authority](https://www.gov.uk/cma-cases/investigation-into-googles-privacy-sandbox-browser-changes#industry-testing). This includes advertising-related domains that are also used for non-advertising purposes.

> Chromewill initially work with Disconnect.me, an industry leader in internet privacy, and implement Disconnect.me's tracker protection lists to identify the scripts and domains categorized as advertising. Disconnect is already used by other browsers for similar purposes on the web.

Reviewer: ```suggestion
Chrome will initially work with Disconnect.me, an industry leader in internet privacy, and implement Disconnect.me's tracker protection lists to identify the scripts and domains categorized as advertising. Disconnect is already used by other browsers for similar purposes on the web.
```

We will apply the following process for registration requests:

> 1. If the third-party origin matches a known advertising domain, including if the origin matches an entry on the disconnect.me [advertising list](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/services.json#L1535C11-L1535C11), then the registration request will be rejected. In general, if a domain is considered an advertising domain, subdomains of that domain will also be considered advertising List entries scoped to a particular pathname, however, will have the pathname removed and match to the exact origin in the entry.

Reviewer: Should it be Disconnect.me or disconnect.me? Either way, it should be consistent.

Author: Updated to consistently capitalise Disconnect and Disconnect.me (where the URL is being a bit more emphasised).

> 1. If the third-party origin matches a known advertising domain, including if the origin matches an entry on the disconnect.me [advertising list](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/services.json#L1535C11-L1535C11), then the registration request will be rejected. In general, if a domain is considered an advertising domain, subdomains of that domain will also be considered advertising List entries scoped to a particular pathname, however, will have the pathname removed and match to the exact origin in the entry.

Reviewer: ```suggestion
1. If the third-party origin matches a known advertising domain, including if the origin matches an entry on the disconnect.me [advertising list](https://github.com/disconnectme/disconnect-tracking-protection/blob/master/services.json#L1535C11-L1535C11), then the registration request will be rejected. In general, if a domain is considered an advertising domain, subdomains of that domain will also be considered advertising list entries scoped to a particular pathname, however, will have the pathname removed and match to the exact origin in the entry.
```

2. Steps to reproduce a broken user-facing experience must be provided. In particular, this should be an experience for the user operating the device where the cookie is stored, and not a user performing later analysis of data. If we cannot validate a broken user experience then the registration request will be rejected.

> 3. Otherwise the registration request will be approved.

Reviewer: Surely point 3 isn't a logical third step? I would say to not use a numbered list here, it does not seem to be a sequence of instructions. It can be two bullet points and then this line as a conclusion.

Author: Converted to bullets, but kept them all in the list just for keeping them visually close and emphasised the outcome at each step, i.e. if it's this: rejected, if it's that: rejected, otherwise: approved.

## Apply for the deprecation trial

{% Aside %}

Before applying, you must report the functionality that will be broken by the third-party cookie deprecation, at [goo.gle/report-3pc-broken](https://goo.gle/report-3pc-broken).

{% endAside %}

Include reproduction steps that our team can use to verify the functional breakage. Alternatively if it’s easier and/or your functionality is gated by login or similar, you can provide a link to a recording of the steps to reproduce the problem, using [Chrome DevTools Recorder](/docs/devtools/recorder/).

1. Starting the week of November 27, 2023 navigate to the "3PCD Deprecation Trial" via the [list of Active Trials on the Chrome Origin Trials page](/origintrials/#/trials/active) and click "Register".

> 2. For "Web Origin" you should provide the origin which serves your embedded page or scripts.

Reviewer: ```suggestion
2. For "Web Origin", provide the origin which serves your embedded page or scripts.
```

> 3. The "Third-party matching" option will depend on how you need to provide the token. The options are explained in more detail in [Add the trial token](#add-the-trial-token) below.

Reviewer: ```suggestion
3. The "Third-party matching" option will depend on how you need to provide the token. The options are explained in more detail in [Add the trial token](#add-the-trial-token).
```

    * If you are providing the token via HTTP header or meta tag on your own embedded pages, **do not check** "Third-party matching".
    * If you are injecting the token via JavaScript into a different site, you **must check** "Third-party matching".
    * If you need to do both, you will need to make separate registrations.
4. If you host cross-site content across multiple subdomains, then tick the "match all subdomains" option.

>     * Tokens will match multiple subdomains similarly to wildcard matching, e.g. `*.<domain>`. Request a token for `example.com` and it can be provided on `a.example.com`, `b.example.com`. Third-party cookie access will still only be re-enabled for the specific origins that provide the token, not all the subdomains. See [What cookies are enabled when subdomain matching is enabled?](#what-cookies-are-enabled-when-subdomain-matching-is-enabled) below.

Reviewer: ```suggestion
    * Tokens will match multiple subdomains similarly to wildcard matching, e.g. `*.<domain>`. Request a token for `example.com` and it can be provided on `a.example.com`, `b.example.com`. Third-party cookie access will still only be re-enabled for the specific origins that provide the token, not all the subdomains. See [What cookies are enabled when subdomain matching is enabled?](#what-cookies-are-enabled-when-subdomain-matching-is-enabled).
```

    * If you host cross-site content across separate origins that are not under the same domain, you will need to make separate registrations for each origin.

> 5. Acknowledge all conditions included in ‘Disclosure and Acknowledgement’ by checking all boxes

Reviewer: ```suggestion
5. Acknowledge all conditions included in ‘Disclosure and Acknowledgement’ by checking all boxes.
```

> 6. Submit request

Reviewer: ```suggestion
6. Submit request.
```

7. We require additional information to process your request. You will receive an email notification with an auto-generated ticket asking for the following:
    * The number of subdomains tied to your requested origin
    * The bug ID or link for the associated third-party breakage repo bugs that you previously reported to [goo.gle/report-3pc-broken](https://goo.gle/report-3pc-broken).

>     * Any additional information/context about the breakage/use case that you would like us to consider. (In cases of an appeal for a denied trial request, explain why/how your origin meets the outlined criteria for this trial)

Reviewer: ```suggestion
    * Any additional information/context about the breakage/use case that you would like us to consider. (In cases of an appeal for a denied trial request, explain why/how your origin meets the outlined criteria for this trial).
```

{% Img src="image/VWw0b3pM7jdugTkwI6Y81n6f5Yc2/fCvvi7tkodM97Yrnl98X.png", alt="Diagram reiterating the token being provided on the page response.", width="800", height="266" %}

If you need cross-site cookies to be on the very first request to your origin in the session, then you can also use `Critical-Origin-Trial` header passing the trial name:

```text
Critical-Origin-Trial: Tpcd
```

This will cause the browser to retry the request with third-party cookies enabled.

The deprecation trial is provided as a persistent trial, which means that once the token has been received by the browser, the trial behavior will be applied until an iframe is loaded without a trial token present. It is recommended to send the trial token on each iframe load consistently.

### Provide the token via meta tag

Within a page, you can use a meta tag in the document `<head>`:

```html
<meta http-equiv="origin-trial" content="TOKEN_GOES_HERE">
```

The meta tag will enable cross-site cookies for subsequent requests or JavaScript in the page, but you will need to use the HTTP header if you require existing cookies to be sent on the initial request.

### Inject the token via JavaScript

> If you need to enable third-party cookies for your origin before or without serving your own page request, e.g. cookies are required on a cross-site image request, or you intend to create an iframe via JavaScript, then you can inject the token into the top-level site using JavaScript:

Reviewer: ```suggestion
If you need to enable third-party cookies for your origin before or without serving your own page request, for example, if cookies are required on a cross-site image request, or you intend to create an iframe via JavaScript, then you can inject the token into the top-level site using JavaScript:
```

Trial deactivation does not take subdomain matching into account. To deactivate the trial an iframe exactly matching the origin in the registration must be loaded without a token. So a registration for `https://test.example` with subdomain matching can only be disabled by an `https://test.example` iframe without a token. This may change in the future, so we recommend providing a token on **all** subframe iframes when you want to enable the trial and removing tokens from **all** iframes when you want to deactivate the trial.

## Frequently Asked Questions

1. What if I have questions about the [Disconnect.me](http://disconnect.me) list?
    * Contact Disconnect at [support@disconnect.me](mailto:support@disconnect.me) as we do not manage the Disconnect list. For more information, see their [tracker protection page](https://disconnect.me/trackerprotection). 
2. Can I register for the deprecation trial if my domain is used for both advertising and non-advertising purposes?

>     * Third-party embeds and services used for advertising are not eligible for the deprecation trial, for the reasons explained in this blog above. This includes advertising-related domains that are also used for non-advertising purposes. For more information please see the [Eligibility criteria and review process](#eligibility-criteria-and-review-process) section above.

Reviewer: ```suggestion
    * Third-party embeds and services used for advertising are not eligible for the deprecation trial, for the reasons explained in this blog previously. This includes advertising-related domains that are also used for non-advertising purposes. For more information, see the [Eligibility criteria and review process](#eligibility-criteria-and-review-process) section.
```

3. Will sites be able to see which one of their partners have enrolled in the deprecation trial? Will they be able to limit the registration across their partners?

>     * Yes, sites can see which embeds and services are relying on a deprecation trial token via the application panel in Chrome Devtools. See [Troubleshoot Chrome origin trials](/docs/web-platform/origin-trial-troubleshooting/) for more info.

Reviewer: ```suggestion
    * Yes, sites can see which embeds and services are relying on a deprecation trial token via the application panel in Chrome Devtools. See [Troubleshoot Chrome origin trials](/docs/web-platform/origin-trial-troubleshooting/) for more information.
```

>     * Top level sites won’t be able to limit registration across their partners or the embeds and services on their page. Please contact the partner if that is desired.

Reviewer: ```suggestion
    * Top level sites won’t be able to limit registration across their partners or the embeds and services on their page. Contact the partner if that is desired.
```

> 4. How is this trial different from other trials like the User Agent Reduction origin trial?

Reviewer: ```suggestion
4. How is this trial different from other trials like the User-Agent reduction origin trial?
```

5. Will there be a first-party deprecation trial for third-party cookie deprecation that top level sites can enroll in to enable 3PCs for their entire site?
    * At the moment we are focusing on third-party embeds and services. We recommend first-party sites continue [making changes to their sites directly to fix the breakage](/blog/cookie-countdown-2023oct/) and encourage their embedded third parties to sign up for this deprecation trial.
6. How long will it take to review my deprecation trial application?  Where can I check on the status of my application?
    * Response times may vary; you are encouraged to begin the registration process as soon as possible to ensure you will be ready ahead of 1% third-party cookie deprecation in early Q1. If you have not received any response within 1-2 weeks of submitting your registration, please contact [3pc-deprecationtrial@google.com](mailto:3pc-deprecationtrial@google.com).
    * Bug thread for open conversation, decision status and rationale.
7. Our deprecation trial registration has been approved, and we've deployed a trial token as recommended. However, the deprecation trial isn't working as expected. What should we do?

>     * [Troubleshoot Chrome origin trials](https://goo.gle/ot-ts) provides a checklist for troubleshooting origin trials. In particular, for this deprecation trial, make sure you have registered for the correct origin, opted for a third-party token if necessary, and correctly provided the token via an HTTP header, meta tag or (for a third-party token) using JavaScript. You can learn more about third-party origin trials at [Third-party origin trials](https://goo.gle/ot-3p), and there is a deprecation trial demo at [Chrome origin trial demo: Token injected by third-party script](https://ot-3p.glitch.me). If you continue to experience problems, please contact [origin-trials-support@google.com](mailto:origin-trials-support@google.com).

Reviewer: ```suggestion
    * [Troubleshoot Chrome origin trials](https://goo.gle/ot-ts) provides a checklist for troubleshooting origin trials. In particular, for this deprecation trial, make sure you have registered for the correct origin, opted for a third-party token if necessary, and correctly provided the token via an HTTP header, meta tag or (for a third-party token) using JavaScript. You can learn more about third-party origin trials at [Third-party origin trials](https://goo.gle/ot-3p), and there is a deprecation trial demo at [Chrome origin trial demo: Token injected by third-party script](https://ot-3p.glitch.me). If you continue to experience problems, contact [origin-trials-support@google.com](mailto:origin-trials-support@google.com).
```
